dask/array/core.py

```diff
@@ -4767,7 +4767,9 @@ def dtype(x):
 
     result = np.empty(shape=shape, dtype=dtype(deepfirst(arrays)))
 
-    for (idx, arr) in zip(slices_from_chunks(chunks), core.flatten(arrays)):
+    for (idx, arr) in zip(
+        slices_from_chunks(chunks), core.flatten(arrays, container=(list, tuple))
```

Comment:
Adding `container=(list, tuple)` to `flatten` here is a driveby. Seemed weird to me that we gave it earlier in the function but not here. Not sure if there's any reason for the inconsistency, so I made it consistent.